# CAR-SALES dataset

Dữ liệu `car-sales.csv`: bộ dữ liệu chứa thông tin các ôtô cũ ở một cửa hàng bán ô tô cũ. 

Bạn hãy thực hiện các yêu cầu sau:


In [2]:
import pandas as pd

In [3]:
def load_data(file: str):
    df = pd.read_csv(f"data/{file}.csv")
    return df

### 1. Đọc dữ liệu và cho biết dữ liệu có bao nhiêu dòng, bao nhiêu cột

Đáp án: (10 dòng, 5 cột)

In [5]:
# Your code here
load_data("car-sales").head()

Manufacturer Colour  Odometer  Doors    Price
0       Toyota  White    150043      4   4000.0
1        Honda    Red     87899      4   5000.0
2       Toyota   Blue     32549      3   7000.0
3          BMW  Black     11179      5  22000.0
4       Nissan  White    213095      4   3500.0

### 2. Cho biết thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng

In [ ]:
# Your code here

### 3. Trong dữ liệu có thông tin của bao nhiêu xe?

Đáp án: 10 xe

In [ ]:
# Your code here

### 4. Đếm số xe theo theo hãng

Đáp án: Toyota: 10 xe, Honda: 3 xe, Nissan: 2 xe, BMW: 1 xe

In [ ]:
# Your code here

### 5. Đếm số xe theo màu


In [ ]:
# Your code here

### 6. Tổng giá trị các xe trong cửa hàng

Đáp án: 76450 USD


In [ ]:
# Your code here

### 7. Top 3 xe có giá cao nhất


In [ ]:
# Your code here

### 8. Top 3 xe đã chạy nhiều cây số (km)nhất (Odometer)

Đáp án:
- Xe Nissan, White, 213095 km
- Xe Toyota, White, 150043 km
- Xe Toyota, Green, 99213 km

In [ ]:
# Your code here

### 9. Khảo sát xe của hãng Honda

- Tính tỷ lệ xe Honda trong dữ liệu của cửa hàng (0.3)
- Cho biết giá trung bình của xe hãng Honda của cửa hàng (6500)
- Cho biết các xe của hãng Honda trong cửa hàng có những màu sắc nào (Red, Blue)

In [ ]:
# Your code here

# 10. Tính toán các giá trị sau của  thuộc tính giá ('Price')

- Giá trung bình các xe (7645)
- Giá trị trung vị các xe (6625)
- Miền giá trị của xe (18500)
- Miền phân vị của xe (2750)
- Điểm phân vị 0.4 và 0.6 của giá ([5750, 7000])

In [ ]:
# Your code here

---

# TIPS dataset

Dữ liệu chứa trong file `tips.csv` chứa thông tin tiền tip của khách hàng tại một nhà hàng. Dữ liệu có các thông tin sau:

- `total_bill`: tổng hoá đơn (USD)
- `tip`: tiền tip/ tiền boa (USD)
- `sex`: giới tính người thanh toán hóa đơn
- `smoker`: có hút thuốc hay không
- `day`: ngày trong tuần
- `time`: thời gian dùng bữa(buổi sáng, buổi trưa, buổi tối)
- `size`: số lượng người tham gia

Bạn hãy thực hiện thống kê mô tả để tìm hiểu về hành vi tip của khách hàng tại nhà hàng

1. Đọc dữ liệu


In [ ]:
# Your code here

2. Xoá cột `Unnamed: 0`


In [ ]:
# Your code here

3. Cho biết dữ liệu có bao nhiêu dòng, bao nhiêu thuộc tính.


In [ ]:
# Your code here

4. Trong các thuộc tính, thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng


In [ ]:
# Your code here

5. Có bao nhiêu nam, bao nhiêu nữ. Tỷ lệ bao nhiêu?


In [ ]:
# Your code here

6. Có bao nhiêu người hút thuốc. Tỷ lệ bao nhiêu?


In [ ]:
# Your code here

7. Ngày nào trong tuần khách hàng hay đến dùng bữa nhất. Tỷ lệ bao nhiêu?


In [ ]:
# Your code here

8. Nhà hàng phục vụ bữa ăn nào nhiều nhất (Breakfast, Lunch, Dinner)?


In [ ]:
# Your code here

9. Nhóm khách hàng dùng bữa tại nhà hàng nhiều nhất là bao nhiêu người?


In [ ]:
# Your code here

10. Trung bình trong một bữa ăn nhà hàng phục vụ mấy khách?

In [ ]:
# Your code here

---